realsense2_camera/src/ros_sensor.cpp

```diff
@@ -153,7 +153,7 @@ void RosSensor::set_sensor_parameter_to_ros(rs2_option option)
 {
     std::string module_name = create_graph_resource_name(rs2_to_ros(get_info(RS2_CAMERA_INFO_NAME)));
     const std::string option_name(module_name + "." + create_graph_resource_name(rs2_option_to_string(option)));
-    float value = get_option(option);
+    int value = get_option(option);
     _params.getParameters()->setRosParamValue(option_name, &value);
```

Comment:
> Can you explain how is this solve the issue?

Here the address of the float variable "value" is getting passed to "setRosParamValue" function.
Since the ROS params (exposure & gain) are of int type, it expects an integer value in that address, and it tries to read an integer value from that address. Since, the representations of float and integer are different, wrong value is read.

> I we do need it I suggest going with
>‘auto value = static_cast(get_option(option))’

Will try this.